src/main/webapp/cdn/blockly/generators/propc/sensors.js

```diff
@@ -1236,6 +1236,120 @@ Blockly.propc.GPS_velocity = function () {
     return [code, Blockly.propc.ORDER_ATOMIC];
 };
 
+Blockly.Blocks.GPS_date_time = {
+    helpUrl: Blockly.MSG_GPS_HELPURL,
+    init: function () {
+        this.setTooltip(Blockly.MSG_GPS_VELOCITY_TOOLTIP);
+        this.setColour(colorPalette.getColor('input'));
+        this.appendDummyInput()
+                .appendField("GPS current ")
+                .appendField(new Blockly.FieldDropdown([
+                    ["year", "GPS_UNIT_YEAR"],
+                    ["month", "GPS_UNIT_MONTH"],
+                    ["day", "GPS_UNIT_DAY"],
+                    ["hour", "GPS_UNIT_HOUR"],
+                    ["minute", "GPS_UNIT_MINUTE"],
+                    ["second", "GPS_UNIT_SECOND"]
+                ], function (unit) {
+                    var zone_label = this.sourceBlock_.getField('ZONE_LABEL');
+                    var zone_value = this.sourceBlock_.getField('ZONE_VALUE');
+                    if (unit === 'GPS_UNIT_HOUR') {
+                        zone_label.setVisible(true);
+                        zone_value.setVisible(true);
+                    } else {
+                        zone_label.setVisible(false);
+                        zone_value.setVisible(false);
+                    }
+                    this.sourceBlock_.render();
+                }), "TIME_UNIT")
+                .appendField("time zone", 'ZONE_LABEL')
+                .appendField(new Blockly.FieldDropdown([
+                    ['UTC+0', '0'],
```

Comment:
This list does not address time zones that are not in whole hour increments. See [Wikipedia](https://en.wikipedia.org/wiki/Time_zone).